lightning/src/ln/reorg_tests.rs

```diff
@@ -0,0 +1,154 @@
+	for i in CHAN_CONFIRM_DEPTH + 2..CHAN_CONFIRM_DEPTH + ANTI_REORG_DELAY - 1 {
+		header = BlockHeader { version: 0x20000000, prev_blockhash: header.bitcoin_hash(), merkle_root: Default::default(), time: 42, bits: 42, nonce: 42 };
+		nodes[1].block_notifier.block_connected_checked(&header, i, &vec![], &[0; 0]);
+		headers.push(header.clone());
+	}
+	check_added_monitors!(nodes[1], 0);
+	assert_eq!(nodes[1].node.get_and_clear_pending_events().len(), 0);
+
+	if claim {
+		// Now reorg back to CHAN_CONFIRM_DEPTH and confirm node 2's broadcasted transactions:
+		for (height, header) in (CHAN_CONFIRM_DEPTH + 1..CHAN_CONFIRM_DEPTH + ANTI_REORG_DELAY - 1).zip(headers.iter()).rev() {
+			nodes[1].block_notifier.block_disconnected(&header, height);
+		}
+
+		header = BlockHeader { version: 0x20000000, prev_blockhash: Default::default(), merkle_root: Default::default(), time: 42, bits: 42, nonce: 42 };
+		nodes[1].block_notifier.block_connected(&Block { header, txdata: claim_txn }, CHAN_CONFIRM_DEPTH + 1);
+
+		// ChannelManager only polls ManyChannelMonitor::get_and_clear_pending_htlcs_updated when we
+		// probe it for events, so we probe non-message events here (which should still end up empty):
+		assert_eq!(nodes[1].node.get_and_clear_pending_events().len(), 0);
+	} else {
+		// Confirm the timeout tx and check that we fail the HTLC backwards
+		header = BlockHeader { version: 0x20000000, prev_blockhash: header.bitcoin_hash(), merkle_root: Default::default(), time: 42, bits: 42, nonce: 42 };
+		nodes[1].block_notifier.block_connected_checked(&header, CHAN_CONFIRM_DEPTH + ANTI_REORG_DELAY, &vec![], &[0; 0]);
+		expect_pending_htlcs_forwardable!(nodes[1]);
+	}
+
+	check_added_monitors!(nodes[1], 1);
+	// Which should result in an immediate claim/fail of the HTLC:
+	let htlc_updates = get_htlc_update_msgs!(nodes[1], nodes[0].node.get_our_node_id());
+	if claim {
+		assert_eq!(htlc_updates.update_fulfill_htlcs.len(), 1);
+		nodes[0].node.handle_update_fulfill_htlc(&nodes[1].node.get_our_node_id(), &htlc_updates.update_fulfill_htlcs[0]);
+	} else {
+		assert_eq!(htlc_updates.update_fail_htlcs.len(), 1);
+		nodes[0].node.handle_update_fail_htlc(&nodes[1].node.get_our_node_id(), &htlc_updates.update_fail_htlcs[0]);
+	}
+	commitment_signed_dance!(nodes[0], nodes[1], htlc_updates.commitment_signed, false, true);
+	if claim {
+		expect_payment_sent!(nodes[0], our_payment_preimage);
+	} else {
+		let events = nodes[0].node.get_and_clear_pending_msg_events();
+		assert_eq!(events.len(), 1);
+		if let MessageSendEvent::PaymentFailureNetworkUpdate { update: HTLCFailChannelUpdate::ChannelClosed { ref is_permanent, .. } } = events[0] {
+			assert!(is_permanent);
+		} else { panic!("Unexpected event!"); }
+		expect_payment_failed!(nodes[0], our_payment_hash, false);
+	}
+}
+
+#[test]
+fn test_onchain_htlc_claim_reorg_local_commitment() {
+	do_test_onchain_htlc_reorg(true, true);
+}
+#[test]
+fn test_onchain_htlc_timeout_delay_local_commitment() {
```

Comment:
I think both local/remote cases are already covered by test_htlc_on_chain_timeout?

Reply:
Hmm, probably. I still like checking that the one-block-difference is exactly the same but does something different. If you really feel strongly I can remove it.